### PR TITLE
[Caffe2 Importer] Add NCHW2NHWC Caffe2 op.

### PR DESCRIPTION
**Pull request by beicy**
*Description*:
https://caffe2.ai/docs/operators-catalogue.html#nchw2nhwc
*Testing*:
Test added.
*Documentation*:
[Optional Fixes #issue]

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: tests/unittests/caffe2ImporterTest.cpp [hunk 1]
```diff
@@ -1341,3 +1341,41 @@ TEST(caffe2, sparseToDense) {
...
+
+  // Destroy the loader after the graph is loaded since the following execution
+  // will not depend on anyting from the loader.
+  {
+    Caffe2ModelLoader caffe2LD(NetDescFilename, NetWeightFilename, {"inputs"},
+                               {&inputs.getType()}, *F);
+    output = caffe2LD.getSingleOutput();
+  }
+
+  // High level check on the content of the graph. We have 1 transpose and 1
```
Comment (jfix71) [inline tests/unittests/caffe2ImporterTest.cpp]: I believe checking the output shape in scope here for these tests, correct? If so can you add a check that the output shape is as expected (`{1,3,4,2}`)?

File: tests/unittests/caffe2ImporterTest.cpp [hunk 2]
```diff
@@ -1341,3 +1341,41 @@ TEST(caffe2, sparseToDense) {
...
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  std::string NetDescFilename(
+      "tests/models/caffe2Models/NCHW2NHWC_predict_net.pbtxt");
+  std::string NetWeightFilename(
+      "tests/models/caffe2Models/empty_init_net.pbtxt");
+
+  Placeholder *output;
+  Context ctx;
```
Comment (nadavrot) [inline tests/unittests/caffe2ImporterTest.cpp]: Why do we need the context?

File: tests/unittests/caffe2ImporterTest.cpp (hunk 1 above)
Comment (beicy) [reply tests/unittests/caffe2ImporterTest.cpp]: Added the shape check.

File: tests/unittests/caffe2ImporterTest.cpp (hunk 2 above)
Comment (beicy) [reply tests/unittests/caffe2ImporterTest.cpp]: It is actually for getting the output shape -- although the check is just added...

**Pull request merged**